server/src/main/java/org/elasticsearch/node/NodeConstruction.java

```diff
@@ -514,7 +514,7 @@ private SettingsModule validateSettings(Settings envSettings, Settings settings,
         // so we might be late here already
         SettingsModule settingsModule = new SettingsModule(
             settings,
-            additionalSettings,
+            additionalSettings.stream().distinct().collect(Collectors.toList()),
```

Comment:
Why is this needed? We should not have duplicate settings

Reply:
Yes, I don't think a change to this file is needed.

Reply:
There is a setting `cluster.logsdb.enabedl` which is used by `StackPlugin` and `LogsDBPlugin`, and which is returned by method `getSettings` by both. SO I have two options here:
1. I deduplicate the additional settings as done here.
2. I return the setting only from one of the plugins.

I saw the code in `NodeConstruction` passing these settings to the `SettingsModule` and I see the logic to register settings which does not allow duplicates. Some of my tests were failing before introducing this change because of the exception thrown while registering duplicate settings. I see the logic uses a list of settings rather than a set which tells me that for some reason duplicates are not expected.

For my understanding would be useful to know why duplicates are not expected.

@rjernst what is the right way to proceed? Is there any reason why we can't return the same settings from two different plugins and deduplicate.

Reply:
> I return the setting only from one of the plugins.

This is the only option. Registering the same setting twice isn't allowed because this would allow for leniency.